### ISSUE TITLE
Cannot open wsltty from imported distributions

### ISSUE THREAD

**Issue by yszhang95** [mintty/wsltty#296]
Sorry to bother. 

I have similar issues to this [one](https://github.com/mintty/wsltty/issues/226) during running CentOS7 and fedora. I got the imported source from this [link](https://github.com/mishamosher/CentOS-WSL) and this [link](https://koji.fedoraproject.org/koji/packageinfo?packageID=26387).

I installed the latest version, 3.5.1 of wsltty. It will create the shortcuts under start menu. Neither one work. I clicked the shortcuts and nothing happens.
![image](https://user-images.githubusercontent.com/43416813/133022342-8ba657d2-f401-45dd-af23-1dd3e6ca6ad1.png)

**Comment by Biswa96** [comment]
Where are the shortcuts pointing to? See the shortcuts properties.

**Comment by yszhang95** [comment]
After checking the directory, `%LOCALAPPDATA%\wsltty`, I find the shortcut. From the properties, I find
```
C:\Users\MYNAME\AppData\Local\wsltty\bin\mintty.exe --WSL="CentOS7" --configdir="C:\Users\MYNAME\AppData\Roaming\wsltty" -~  -
```
And `wsl -l` gives me
```
Windows Subsystem for Linux Distributions:
CentOS7 (Default)
docker-desktop
fedora
docker-desktop-data
```

**Comment by Biswa96** [comment]
Paste this in Win+R, does it work?

```
C:\Users\MYNAME\AppData\Local\wsltty\bin\mintty.exe --WSL= --configdir="C:\Users\MYNAME\AppData\Roaming\wsltty" -~  -
```

**Comment by yszhang95** [comment]
It works! Thanks for your kind reply! I am happy to close this issue. Or do you want to keep it open and enhance wsltty later?

**Comment by Biswa96** [comment]
I would like to wait for @mintty comment. I am not familiar with the shortcuts creation here. Though I did not find any issue in your provided command.

**Comment by yszhang95** [comment]
It seems `mintty` cannot recognize the option `--WSL=CentOS7`. Following you suggestion, I changed to `--WSL=`, it works. With `--WSL=CentOS7`, `Win+R` cannot work either.

**Comment by mintty** [comment]
Works here with CentOS8.

**Comment by yszhang95** [comment]
Do you have multiple Linux distributions? If so, it seems this issue cannot be reproduced. Feel free to close it. One follow-up question is how to remove the shortcuts from start menu, since they do not work for me? Simply delete shortcuts under
```
C:\Users\MYNAME\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\
```
?

**Comment by mintty** [comment]
You have multiple distributions yourself, CentOS7 and fedora. Can you start wsltty with the "fedora Terminal" shortcut?
And yes, you can simply delete shortcuts.

**Comment by yszhang95** [comment]
It does not work either. I can only open the wsltty by leaving `WSL=` as empty. It will start the default wsl distribution. I can close this issue if this behavior cannot be reproduced. I am sorry I cannot fix this issue by myself because of lack of knowledge. Many thanks for the help from @mintty and @Biswa96 so far!

**Comment by robolivable** [comment]
It seems as though the output is the same whether or not I specify `--WSL="Ubuntu"`:
```
[rldeo@epislon ~]$ /cygdrive/c/Users/rldeo/AppData/Local/wsltty/bin/mintty.exe -RW --WSL="Ubuntu" --configdir="C:\Users\rldeo\AppData\Roaming\wsltty" -~
WSL distribution name Ubuntu
-- guid {f2d50f5b-c3c9-4208-bdfe-05adca36f054}
-- flag 15
-- root C:\Users\rldeo\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\rootfs
-- pack CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc
-- full CanonicalGroupLimited.UbuntuonWindows_2004.2020.424.0_x64__79rhkp1fndgsc
-- icon C:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_2004.2020.424.0_x64__79rhkp1fndgsc\images\icon.ico
[rldeo@epislon ~]$ /cygdrive/c/Users/rldeo/AppData/Local/wsltty/bin/mintty.exe -RW --WSL= --configdir="C:\Users\rldeo\AppData\Roaming\wsltty" -~
WSL distribution name Ubuntu
-- guid {f2d50f5b-c3c9-4208-bdfe-05adca36f054}
-- flag 15
-- root C:\Users\rldeo\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\rootfs
-- pack CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc
-- full CanonicalGroupLimited.UbuntuonWindows_2004.2020.424.0_x64__79rhkp1fndgsc
-- icon C:\Program Files\WindowsApps\CanonicalGroupLimited.UbuntuonWindows_2004.2020.424.0_x64__79rhkp1fndgsc\images\icon.ico
[rldeo@epislon ~]$
```

FWIW, running in Cygwin also lets me see the seg fault:
```
[rldeo@epislon ~]$ /cygdrive/c/Users/rldeo/AppData/Local/wsltty/bin/mintty.exe --WSL="Ubuntu" --configdir="C:\Users\rldeo\AppData\Roaming\wsltty" -~
Segmentation fault (core dumped)
[rldeo@epislon ~]$ cat mintty.exe.stackdump
Exception: STATUS_ACCESS_VIOLATION at rip=001801F38F2
rax=0000000000000000 rbx=0000000000000000 rcx=0000000000000000
rdx=000000080003A840 rsi=0000000000000000 rdi=00000000FFFFC4C0
r8 =0000000000000000 r9 =0000000000000059 r10=0000000000000008
r11=0000000000000246 r12=0000000000000248 r13=00000001004CA560
r14=00000000FFFFC520 r15=00000000FFFFC4B0
rbp=00000000FFFFC550 rsp=00000000FFFFC458
program=C:\Users\rldeo\AppData\Local\wsltty\bin\mintty.exe, pid 1233, thread main
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B
Stack trace:
Frame        Function    Args
000FFFFC550  001801F38F2 (000FFFFC518, 00000000000, 00000000012, 00000000000)
000FFFFC550  00100445539 (0018023A538, 0018023A53C, 001004E0C80, 0010049AD88)
000FFFFC690  00100445658 (00800001AD0, 0080003A830, 0080003A8A0, 00000000000)
000FFFFC690  0010045D680 (0018037DFD1, 000FFFFCDF0, 00000000030, 8080808080808080)
000FFFFCCE0  00180049EFD (00000000000, 00000000000, 00000000000, 00000000000)
000FFFFFFF0  00180047856 (00000000000, 00000000000, 00000000000, 00000000000)
000FFFFFFF0  00180047904 (00000000000, 00000000000, 00000000000, 00000000000)
End of stack trace
[rldeo@epislon ~]$
```

**Comment by Biswa96** [comment]
> could this be related to #297 or #302?

Not likely related to wslbridge2. Because OP already confirmed that `--WSL=` option works but `--WSL=fedora` does not. Those issue occurs only in Windows 10 build 17763 or before (probably LTSC versions).

**Comment by yszhang95** [comment]
Sorry for the late response. This update fixes the issue for me. I cloned the latest `mintty` repository and compile under `cygwin`. And then replaced the `mintty.exe` with the bleeding edge version. Now I can run `mintty.exe` with the options `--WSL="CentOS7"`.

BTW, I am sorry that I made a mistake during the last post. I did not copy the `wslbridge2.exe` to the directory `/bin` under `cygwin` environment. I am not sure if the error messages were relevant to the issue that you fixed. I am truly sorry about this if the meessages mislead you.

Many thanks for fixing this issue! It is really great to switch the WSL distributions under `mintty`.
